### PR TITLE
feat: add iron bank Dockerfile & manifest

### PR DESCRIPTION
**Pull request by ericpaulsen**
adds Iron Bank-compatible Dockerfile & hardening manifest. this will be the master copy that we will copy into the Iron Bank repo.

### CODE REVIEW

**Review by kylecarbs — DISMISSED** [review]
I don't want this to be a top-level folder. Let's put it inside of `scripts`.

File: ironbank/hardening_manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  # Needs to be kept in sync with the resource below.
+  TERRAFORM_CODER_PROVIDER_VERSION: "0.6.9"
+
+# Docker image labels
+labels:
+  org.opencontainers.image.title: "coder-service-v2"
+  # Human-readable description of the software packaged in the image
+  org.opencontainers.image.description: "Coder server binary, includes REST API, Terraform, and dashboard"
+  # License(s) under which contained software is distributed
+  org.opencontainers.image.licenses: "APGL"
```
Comment (deansheather) [inline ironbank/hardening_manifest.yaml]: AGPL* and also enterprise licensed, maybe we should just remove the licenses field?

File: ironbank/hardening_manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    value: 2c88555777f1d9cc77a8f049093f4002472dc43d52b026e6784ef477bdced4a2
+# Terraform binary, bundled inside of Coder to support air-gapped installs.
+- url: https://releases.hashicorp.com/terraform/1.3.7/terraform_1.3.7_linux_amd64.zip
+  filename: "terraform.zip"
+  validation:
+    type: sha256
+    value: b8cf184dee15dfa89713fe56085313ab23db22e17284a9a27c0999c67ce3021e
+# Coder Terraform provider, bundled inside of Coder to support air-gapped
+# installs.
+- url: https://github.com/coder/terraform-provider-coder/releases/download/v0.6.9/terraform-provider-coder_0.6.9_linux_amd64.zip
```
Comment (deansheather) [inline ironbank/hardening_manifest.yaml:51]: latest is 0.6.10

**Comment by deansheather** [comment]
`scripts` doesn't seem appropriate as there are no scripts, but we don't have a `deploy` folder like v1....

**Comment by kylecarbs** [comment]
@deansheather I don't think there's a point in creating a new folder type for this because it's so bespoke.

**Comment by ericpaulsen** [comment]
prettier is returning the following error:

```
site/src/components/AuthProvider/AuthProvider.tsx[error] site/src/components/AuthProvider/AuthProvider.tsx: SyntaxError: '>' expected. (23:18)
[error]   21 |
[error]   22 | type UseAuthReturnType = ReturnType<
[error] > 23 |   typeof useActor<AuthContextValue["authService"]>
[error]      |                  ^
[error]   24 | >
[error]   25 |
[error]   26 | export const useAuth = (): UseAuthReturnType => {
```

@deansheather let me know the best path forward to resolve.

**Comment by deansheather** [comment]
@ericpaulsen I'd try checking out `main`, pulling, then checkout the PR branch again and do `git merge main` to update the branch since it's not a file you touched

File: scripts/ironbank/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ARG TERRAFORM_BIN_DIR=/opt/terraform
+RUN mkdir -p "$TERRAFORM_BIN_DIR"
+ARG TERRAFORM_BIN_ZIP=terraform.zip
+COPY "$TERRAFORM_BIN_ZIP" /tmp/terraform.zip
+RUN unzip /tmp/terraform.zip -d "$CODER_BIN_DIR" && \
+	  rm /tmp/terraform.zip
+ENV PATH="${TERRAFORM_BIN_DIR}:${PATH}"
+
+# Install the Coder Terraform provider to a well-known location.
+ARG TERRAFORM_PLUGINS_DIR=/opt/terraform/plugins
```
Comment (kylecarbs) [inline scripts/ironbank/Dockerfile:63]: Do we have to install any other providers for them?

Comment (kylecarbs) [reply scripts/ironbank/Dockerfile:63]: Maybe Kubernetes or something?

Comment (ericpaulsen) [reply scripts/ironbank/Dockerfile:63]: nope - any other Terraform providers they need is up to them. they can install those providers from a self-hosted provider registry.

Comment (kylecarbs) [reply scripts/ironbank/Dockerfile:63]: Coolio

**Comment by deansheather** [comment]
We should merge this even with all of the vulnerabilities reported by trivy, and focus on fixing them after merge. Some of them are vulnerabilities in the base image which we might not be able to resolve.

**Review by sreya — APPROVED** [review]

**Pull request merged**